prbench-bilevel-planning/experiments/gen_multi_run_stats.py

```diff
@@ -0,0 +1,186 @@
+        --metrics success steps planning_time
+"""
+
+import argparse
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+
+
+def calculate_multi_run_stats(
+    base_dir: Path,
+    metrics: list[str] | None = None,
+    output_file: Path | None = None,
+) -> pd.DataFrame:
+    """Calculate mean and std statistics across multiple seed runs.
+
+    Args:
+        base_dir: Base directory containing seed_* subdirectories
+        metrics: List of metric columns to aggregate (if None, uses all numeric columns)
+        output_file: Path to save statistics CSV
+            (if None, saves to base_dir/results_summary.csv)
+
+    Returns:
+        DataFrame with mean and std for each metric
+    """
+    # Find all seed_* directories
+    seed_dirs = sorted(base_dir.glob("seed_*"))
+    if not seed_dirs:
+        raise ValueError(f"No seed_* directories found in {base_dir}")
+
+    print(f"Found {len(seed_dirs)} seed directories: {[d.name for d in seed_dirs]}")
+
+    # Load all results.csv files
+    all_results = []
+    for seed_dir in seed_dirs:
+        results_file = seed_dir / "results.csv"
+        if not results_file.exists():
+            print(f"Warning: {results_file} not found, skipping...")
+            continue
+
+        df = pd.read_csv(results_file)
+        all_results.append(df)
+
+    if not all_results:
+        raise ValueError(f"No valid results.csv files found in {base_dir}/seed_*")
+
+    print(f"Loaded {len(all_results)} results files")
+
+    # Determine which columns to aggregate
+    if metrics is None:
+        # Use all numeric columns except eval_episode (which is an index)
+        sample_df = all_results[0]
+        metrics = [
+            col
+            for col in sample_df.columns
+            if col != "eval_episode" and pd.api.types.is_numeric_dtype(sample_df[col])
+        ]
+        # Handle success as numeric (True=1, False=0)
+        if "success" in sample_df.columns:
+            if "success" not in metrics:
+                metrics.insert(0, "success")
+
+    print(f"Aggregating metrics: {metrics}")
+
+    # Convert success column to numeric if it exists
+    for df in all_results:
+        if "success" in df.columns and df["success"].dtype == "object":
+            df["success"] = df["success"].map(
+                {"True": 1, True: 1, "False": 0, False: 0}
+            )
+
+    # Check if success column exists
+    has_success = "success" in all_results[0].columns
+
+    # Calculate statistics across all episodes in all seeds
+    results: dict[str, list] = {"metric": [], "mean": [], "std": []}
+
+    for metric in metrics:
+        if metric == "success" or not has_success:
+            # For success metric (or if no success column), calculate over ALL episodes
+            all_values = np.concatenate([df[metric].values for df in all_results])
+            mean_val = np.mean(all_values)
+            std_val = np.std(all_values, ddof=1)  # Sample std
```

Comment:
we should make sure that everyone is doing the same thing. and in the end, we should directly compare approaches with as much shared code as possible to avoid possible subtle differences between them. cc @yixuanhuang98